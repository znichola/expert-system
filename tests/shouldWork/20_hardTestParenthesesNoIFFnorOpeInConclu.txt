# Expert System Test Case - Complex Parentheses
# No operations in conclusions, no iff cases

(A + C) <=> E             # A and C implies E
A => B                    # A implies B
C => D                    # C implies D  
(C | D) => F              # C or D implies F
(E + F) => G              # E and F implies G
(!A | B) => H             # not A or B implies H
A + !C <=> I            # A and not C implies I
(A + B) | C => J        # (A and B) or C implies J
!(E + F) | G => K       # not(E and F) or G implies K
(H + I) ^ J => L        # (H and I) xor J implies L
A ^ B => M              # A xor B implies M
!G + H => N             # not G and H implies N
(I | J) + !K => O       # (I or J) and not K implies O
M ^ (N + O) => P        # M xor (N and O) implies P
(A + B) | (!C + D) => Q # (A and B) or (not C and D) implies Q

=E     # Initial facts: E is true

?ABDEFGHIJKLMNOPQ    # Query all derived facts
# Expected:
# True: A,B,C,D,E,F,G,H,J,K,Q
# False: I, L, M, N, O, P