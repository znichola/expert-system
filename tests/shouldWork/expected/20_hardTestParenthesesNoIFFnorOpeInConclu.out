A is True
B is True
D is True
E is True
F is True
G is True
H is True
I is False
J is True
K is True
L is False
M is False
N is False
O is False
P is False
Q is True
